### ISSUE TITLE
Adding Sparkle to manage updates

### ISSUE THREAD

**Issue by FivePixels** [ilime/GoalHub#6]
https://github.com/sparkle-project/Sparkle

Sparkle is an awesome software update framework we can use to push updates quickly and efficiently to users without requiring them to grab the latest release package (the framework does that for them!)

I'll start working on implementation if this is something you're interested in.

**Comment by g1eny0ung** [comment]
Yes, definitely agree with u. Looking forward to your results. 👍

**Comment by FivePixels** [comment]
When I've tried to reclone the repository, I can't seem to compile it. I keep getting the following error:


`../Build/Products/Debug/GoalHub.app: code object is not signed at all`
`In subcomponent: ../Build/Products/Debug/GoalHub.app/Contents/Library/LoginItems/LaunchAtLoginHelper.app`
`Command CodeSign failed with a nonzero exit code`

Do you have any ideas?

**Comment by g1eny0ung** [comment]
Maybe as you think about it.

Sometimes I also have some problems with `LaunchAtLoginHelper`. For example, when I want to release a production build, I have to clear the existed build first, if not the `LaunchAtLoginHelper` will occur an error. 😢 

I agree with u. These problems have a big chance related to Carthage.
